style/ruby/README.md

```diff
@@ -162,3 +168,127 @@ name = "Bozhidar"
 sarcasm = 'I "like" it.'
 ```
 
+# Nomenclatura
+## Métodos Booleanos com Ponto de Interrogação
+
+Os nomes dos métodos predicados (métodos que retornam um valor booleano) devem terminar com um ponto de interrogação (ou seja, `Array#empty?`). Métodos que não retornam um booleano não devem terminar em ponto de interrogação.
+
+## Prefixo de Métodos Booleanos
+
+Evite prefixar métodos predicados com verbos auxiliares como `is`, `does` ou `can`. Essas palavras são redundantes e inconsistentes com o estilo dos métodos booleanos na biblioteca principal do Ruby, como `empty?` e `include?`.
+
+```ruby
+# ruim
+class Person
+  def is_tall?
+    true
+  end
+
+  def can_play_basketball?
+    false
+  end
+
+  def does_like_candy?
+    true
+  end
+end
+
+# bom
+class Person
+  def tall?
+    true
+  end
+
+  def basketball_player?
+    false
+  end
+
+  def likes_candy?
+    true
+  end
+end
+```
+
+## Capitalização
+
+1. **snake_case** 
+* Para métodos e variáveis
+* Para nomear diretórios, por exemplo: `lib/hello_world/hello_world.rb`.
+* Para nomear arquivos, por exemplo: `hello_world.rb`.
+
+2. Uma classe por arquivo
+* Procure ter apenas uma única classe/módulo por arquivo de origem. Nomeie o nome do arquivo como classe/módulo, mas substituindo **CapitalCase** por **snake_case**.
+
+3. **CamelCase** 
+* Para Classes e módulos. (Mantenha siglas como HTTP, RFC, XML em maiúsculas.)
+
+```ruby
+# ruim
+class Someclass
+  # some code
+end
+
+class Some_Class
+  # some code
+end
+
+class SomeXml
+  # some code
+end
+
+class XmlSomething
+  # some code
+end
+
+# bom
+class SomeClass
+  # some code
+end
+
+class SomeXML
+  # some code
+end
+
+class XMLSomething
+  # some code
+end
+```
+
+4. **SCREAMING_SNAKE_CASE**
+* Para outras constantes.
+
+```ruby
+# ruim
+SomeConst = 5
+
+# bom
+SOME_CONST = 5
+```
+
+## Variável de instância memorizada
+
+A variável de instância memorizada deve corresponder ao nome do método, para evitar confusão e bugs.
+
+Em Ruby, a utilização de variáveis de instância memorizadas é comum em situações em que o cálculo ou a recuperação de um valor é custoso em termos de processamento ou tempo, mas o valor permanece constante ao longo do tempo. Em vez de recalcular o valor cada vez que ele é necessário, a variável é armazenada em cache na instância da classe e recuperada quando necessário.
```

Comment:
> O uso de variáveis de instância memorizadas pode melhorar significativamente o desempenho do código em situações em que cálculos ou operações de leitura são custosas e o valor permanece constante. No entanto, é importante lembrar de invalidar o cache da variável quando o valor pode mudar, para garantir que o valor armazenado em cache seja atualizado corretamente.
https://www.justinweiss.com/articles/4-simple-memoization-patterns-in-ruby-and-one-gem/